docs/tutorials/gluon/fit_api_tutorial.md

```diff
@@ -0,0 +1,208 @@
+est = estimator.Estimator(net=resnet_18_v1, 
+                                loss=loss_fn, 
+                                metrics=train_acc, 
+                                trainers=trainer, 
+                                context=ctx)
+
+# Magic line
+est.fit(train_data=train_data_loader,
+              epochs=num_epochs, 
+              batch_size=batch_size)
+```
+
+    [Epoch 0] [Step 256/60000] time/step: 1.420s accuracy: 0.0938 softmaxcrossentropyloss0: 2.9419 <!--notebook-skip-line-->
+    .... <!--notebook-skip-line-->
+    [Epoch 0] finished in 51.375s: train_accuracy: 0.7916 train_softmaxcrossentropyloss0: 0.5750 <!--notebook-skip-line-->
+    [Epoch 1] [Step 256/60000] time/step: 0.414s accuracy: 0.8555 softmaxcrossentropyloss0: 0.3621 <!--notebook-skip-line-->
+    .... <!--notebook-skip-line-->
+    [Epoch 1] finished in 49.889s: train_accuracy: 0.8854 train_softmaxcrossentropyloss0: 0.3157 <!--notebook-skip-line-->
+
+
+### Advanced Usage
+
+Fit API is also customizable with several `Event Handlers` which give a fine grained control over the steps in training and exposes callback methods for : `train_begin`, `train_end`, `batch_begin`, `batch_end`, `epoch_begin` and `epoch_end`.
+
+One can use built-in event handlers such as ```LoggingHandler```, ```CheckpointHandler``` or ```EarlyStoppingHandler``` or to create a custom handler, one can create a new class by inherinting [```EventHandler```](https://github.com/apache/incubator-mxnet/blob/fit-api/python/mxnet/gluon/estimator/event_handler.py#L31).
+
+
+```python
+# Let's reset the model, trainer and accuracy objects from above
+
+resnet_18_v1.initialize(force_reinit=True, init = mx.init.Xavier(), ctx=ctx)
+trainer = gluon.Trainer(resnet_18_v1.collect_params(), 
+                        'sgd', {'learning_rate': learning_rate})
+train_acc = mx.metric.Accuracy()
+
+```
+
+
+```python
+# Define the estimator, by passing to it the model, loss function, metrics, trainer object and context
+est = estimator.Estimator(net=resnet_18_v1,
+                                loss=loss_fn,
+                                metrics=train_acc,
+                                trainers=trainer, 
+                                context=ctx)
+
+# Define the handlers, let's say Checkpointhandler
+checkpoint_handler = event_handler.CheckpointHandler(estimator=est,
+                                                     filepath='./my_best_model.params',
+                                                     monitor='train_accuracy', # Monitors a metric
+                                                     save_best_only=True) # Save the best model in terms of 
+                                                                         # training accuracy
+
+# Magic line
+est.fit(train_data=train_data_loader,
+              epochs=num_epochs,
+              event_handlers=[checkpoint_handler], # Add the event handlers
```

Comment:
can the list wrapper be avoided? if just want to pass a single handler.
seems to be inconsistent with `trainers` where that wasn't a list.

Reply:
Okay. Done.